driver_go18_test.go

```diff
@@ -796,3 +796,17 @@ func TestRowsColumnTypes(t *testing.T) {
 		})
 	}
 }
+
+func TestValuerWithValueReceiverGivenNilValue(t *testing.T) {
+	runTests(t, dsn, func(dbt *DBTest) {
+		dbt.mustExec("CREATE TABLE test (value VARCHAR(255))")
+
+		defer func() {
+			if r := recover(); r != nil {
+				dbt.Errorf("Failed to check typed nil before calling valuer with value receiver")
+			}
+		}()
```

Comment:
Is this recover really necessary?
I don't like it.

Reply:
I can leave the test to panic on failure instead - like below.  If you prefer that I will alter the PR.
```golang
func TestValuerWithValueReceiverGivenNilValue(t *testing.T) {
	runTests(t, dsn, func(dbt *DBTest) {
		dbt.mustExec("CREATE TABLE test (value VARCHAR(255))")
		dbt.db.Exec("INSERT INTO test VALUES (?)", (*testValuer)(nil))
		// the insert will panic if typed nil is not handled before calling a Valuer with a value receiver
	})
}
```

Reply:
Yes, please.  It is much informative than manual recovery:

```
$ go test .
--- FAIL: TestPanic (0.00s)
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 5 [running]:
testing.tRunner.func1(0xc4200900f0)
	/usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x1110920, 0x11d96f0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
_/Users/inada-n/tmp/go-panic-test.TestPanic(0xc4200900f0)
	/Users/inada-n/tmp/go-panic-test/panic_test.go:9 +0x1b
testing.tRunner(0xc4200900f0, 0x113cb30)
	/usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x2de
FAIL	_/Users/inada-n/tmp/go-panic-test	0.009s
```